internal/services/containerapps/container_app_resource.go

```diff
@@ -0,0 +1,435 @@
+
+			existing, err := client.Get(ctx, id)
+			if err != nil {
+				if !response.WasNotFound(existing.HttpResponse) {
+					return fmt.Errorf("checking for presence of existing %s: %+v", id, err)
+				}
+			}
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			envId, err := managedenvironments.ParseManagedEnvironmentID(app.ManagedEnvironmentId)
+			if err != nil {
+				return fmt.Errorf("parsing Container App Environment ID for %s: %+v", id, err)
+			}
+
+			env, err := environmentClient.Get(ctx, *envId)
+			if err != nil {
+				return fmt.Errorf("reading %s for %s: %+v", *envId, id, err)
+			}
+
+			containerApp := containerapps.ContainerApp{
+				Name:     pointer.To(app.Name),
+				Location: location.Normalize(env.Model.Location),
+				Properties: &containerapps.ContainerAppProperties{
+					Configuration: &containerapps.Configuration{
+						Ingress:    helpers.ExpandContainerAppIngress(app.Ingress, id.ContainerAppName),
+						Dapr:       helpers.ExpandContainerAppDapr(app.Dapr),
+						Secrets:    helpers.ExpandContainerSecrets(app.Secrets),
+						Registries: helpers.ExpandContainerAppRegistries(app.Registries),
+					},
+					ManagedEnvironmentId: pointer.To(app.ManagedEnvironmentId),
+					Template:             helpers.ExpandContainerAppTemplate(app.Template, metadata),
+				},
+				Tags: tags.Expand(app.Tags),
+			}
+
+			ident, err := identity.ExpandSystemAndUserAssignedMapFromModel(app.Identity)
+			if err != nil {
+				return err
+			}
+			containerApp.Identity = pointer.To(identity.LegacySystemAndUserAssignedMap(*ident))
+
+			revisionMode := containerapps.ActiveRevisionsMode(app.RevisionMode)
+			containerApp.Properties.Configuration.ActiveRevisionsMode = &revisionMode
+
+			if err := client.CreateOrUpdateThenPoll(ctx, id, containerApp); err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			metadata.SetID(id)
+
+			return nil
+		},
+	}
+}
+
+func (r ContainerAppResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.ContainerApps.ContainerAppClient
+
+			id, err := containerapps.ParseContainerAppID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			existing, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(existing.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+				return fmt.Errorf("reading %s: %+v", *id, err)
+			}
+
+			var state ContainerAppModel
+
+			state.Name = id.ContainerAppName
+			state.ResourceGroup = id.ResourceGroupName
+
+			if model := existing.Model; model != nil {
+				state.Location = location.Normalize(model.Location)
+				state.Tags = tags.Flatten(model.Tags)
+				ident, err := identity.FlattenSystemAndUserAssignedMapToModel(pointer.To(identity.SystemAndUserAssignedMap(*model.Identity)))
```

Comment:
this seems like we're missing some methods for the typed expand/flatten functions for the legacy identity types in `hashicorp/go-azure-helpers`?

> func ExpandLegacySystemAndUserAssignedMapFromModel(input []ModelSystemAssignedUserAssigned) (*LegacySystemAndUserAssignedMap, error)
> func FlattenLegacySystemAndUserAssignedMapToModel(input *LegacySystemAndUserAssignedMap) (*[]ModelSystemAssignedUserAssigned, error)